pkg/metricsadapter/adapter.go

```diff
@@ -18,9 +22,13 @@ type MetricsAdapter struct {
 func NewMetricsAdapter(controller *MetricsController, customMetricsAdapterServerOptions *options.CustomMetricsAdapterServerOptions) *MetricsAdapter {
 	adapter := &MetricsAdapter{}
 	adapter.CustomMetricsAdapterServerOptions = customMetricsAdapterServerOptions
-
+	kubeClient := kubernetes.NewForConfigOrDie(controller.restConfig)
+	kubeFactory := informers.NewSharedInformerFactory(kubeClient, 0)
+	kubeFactory.Core().V1().Secrets().Informer()
+	kubeFactory.Start(context.TODO().Done())
+	kubeFactory.WaitForCacheSync(context.TODO().Done())
```

Comment:
We can construct and start this factory in the controller.

Reply:
Updated.